public/js/draw.js

```diff
@@ -67,14 +66,19 @@ function drawFrame(canvas, video, combinedFeatures) {
   }
 
   // Render HandPose
-  let handPoints;
-  if (handPoses !== undefined && handPoses.length > 0) {
-    handPoints = handPoses[0].landmarks;
-    const handAnnotations = handPoses[0].annotations;
-    drawPoints(ctx, handPoints, 3);
-    Object.entries(handAnnotations).forEach(([, points]) =>
-      drawPath(ctx, points)
-    );
+  let handPoints = null;
+  if (combinedFeatures[1] !== null) {
+    const handPoses = combinedFeatures[1];
+
+    // TODO Remove/merge this to the upper branch to avoid unnecessary complexity and computation
+    if (handPoses !== undefined && handPoses.length > 0) {
```

Comment:
I'm not sure if the `undefined` checks are needed; **are the outputs from respectively `faceMesh.estimateFaces(video)` and `handPose.estimateHands(video)` likely to be `undefined`?**
I was thinking of refactoring that snippet to something like (if `undefined values weren't possible in this case):
```js
if (combinedFeatures[1] !== null) {
  const handPoses = combinedFeatures[1];
  handPoints = handPoses[0].landmarks;
  const handAnnotations = handPoses[0].annotations;
  drawPoints(ctx, handPoints, 3);
  Object.entries(handAnnotations).forEach(([, points]) =>
    drawPath(ctx, points)
  );
}
```